### PR TITLE
Add force permission option

### PR DESCRIPTION
**Pull request by bastienpaulfr**
Hi

I have added a "force" option to ask for permission even if the system tells that the permission is granted. 

I have the need of this option because I am developing a system app. Android is telling me that I got the permission whereas I still need to ask for explicitly. 

You have made a great job ! Tks

### CODE REVIEW

**Comment by pedrovgs** [comment]
Hi @bastienpaulfr 

Could you provide some documentation about the scenario you mention in the PR description?

> I have the need of this option because I am developing a system app. Android is telling me that I got the permission whereas I still need to ask for explicitly.

I'd like to know more about this feature before to merge this PR.

File: dexter/src/main/java/com/karumi/dexter/DexterInstance.java [hunk 1]
```diff
@@ -194,6 +196,9 @@ private PermissionStates getPermissionStates(Collection<String> pendingPermissio
    * https://github.com/Karumi/Dexter/issues/86
    */
   private int checkSelfPermission(Activity activity, String permission) {
+    if (forcePermissionRequest) {
+      return PackageManager.PERMISSION_DENIED;
```
Comment (pedrovgs) [inline dexter/src/main/java/com/karumi/dexter/DexterInstance.java]: Is this the expected behavior? If you ask for a permission and you are trying to force it you always return ``PERMISSION_DENIED``?

File: dexter/src/test/java/com/karumi/dexter/DexterInstanceTest.java [hunk 2]
```diff
@@ -70,20 +70,21 @@
   }
```
Comment (pedrovgs) [inline dexter/src/test/java/com/karumi/dexter/DexterInstanceTest.java]: We should cover the new scenario with tests and update the project documentation. Right now it's not obvious this ``shouldForcePremission`` param should be used if you are developing a system app.

**Comment by bastienpaulfr** [comment]
Hi @pedrovgs 

I prefer adding a force option than checking that the app is system app. May be on some devices the original logic is working without my changes. Then the force option would be not necessary. 

I will provide some documentation about this request and submit it again. I will also divide all my requests to make things cleaner. Thank you for the review.

File: dexter/src/main/java/com/karumi/dexter/DexterInstance.java (hunk 1 above)
Comment (bastienpaulfr) [inline dexter/src/main/java/com/karumi/dexter/DexterInstance.java]: According to the logic flow, when `PERMISSION_DENIED` is returned, then the permission is explicitly requested by `AndroidPermissionService` class.

**Comment by bastienpaulfr** [comment]
I will close this pull request to make new ones from separate branches

**Pull request closed without merging**